res/controllers/Gemini-G4V-scripts.js

```diff
@@ -0,0 +1,562 @@
+            this.inKey = "slip_enable";
+        },
+    });
+    this.fxOnOffBtn = new components.Button({
+        midi: [0x8F + midiChannel, 0x1A],
+        group: "[EffectRack1_EffectUnit"+script.deckFromGroup(this.currentDeck)+"]",
+        key: "group_"+this.currentDeck+"_enable",
+        type: components.Button.prototype.types.toggle,
+    });
+    // Loop Move is a pot not an encoder so requires special treatment
+    this.loopMovePot = new components.Component({
+        key: "loop_move",
+        oldValue: 0,
+        input: function(_channel, _control, value, _status, _group) {
+            this.inSetParameter(value-this.oldValue);
+            this.oldValue = value;
+        },
+    });
+    // Sample Vol is a pot not an encoder so requires special treatment
+    this.sampleVolPot = new components.Component({
+        key: "loop_move",
+        input: function(_channel, _control, value, _status, _group) {
+            for (let i = 1; i < 65; i++) {
+                engine.setValue("[Sampler"+i+"]", "volume", value/0x7f);
+            }
+        },
+    });
+    this.fxSuperPot = new components.Pot({
+        group: "[EffectRack1_EffectUnit"+script.deckFromGroup(this.currentDeck)+"]",
+        key: "super1",
+    });
+    this.fxMixPot = new components.Pot({
+        group: "[EffectRack1_EffectUnit"+script.deckFromGroup(this.currentDeck)+"]",
+        key: "mix",
+    });
+
+    // Layers for pads
+    this.padsCue = new components.ComponentContainer();
+    for (let i = 1; i <= 8; i++) {
+        this.padsCue["pad"+i] = new components.HotcueButton({
+            midi: [0x8F + midiChannel, 0x08 + i],
+            group: this.currentDeck,
+            number: i,
+        });
+    }
+    this.padsLoop = new components.ComponentContainer();
+    for (let i = 1; i <= 8; i++) {
+        this.padsLoop["pad"+i] = new components.Button({
+            group: this.currentDeck,
+            midi: [0x8F + midiChannel, 0x08 + i],
+            inKey: "beatloop_"+(0.125*Math.pow(2, i-1))+"_toggle",
+            outKey: "beatloop_"+(0.125*Math.pow(2, i-1))+"_enabled",
+        });
+    }
+    this.padsLoopRoll = new components.ComponentContainer();
+    for (let i = 1; i <= 8; i++) {
+        this.padsLoopRoll["pad"+i] = new components.Button({
+            group: this.currentDeck,
+            midi: [0x8F + midiChannel, 0x08 + i],
+            inKey: "beatlooproll_"+(0.125*Math.pow(2, i-1))+"_activate",
+            outKey: "beatloop_"+(0.125*Math.pow(2, i-1))+"_enabled",
+        });
+    }
+    this.padsBeatjump = new components.ComponentContainer();
+    for (let i = 1; i <= 8; i++) {
+        this.padsBeatjump["pad"+i] = new components.Button({
+            // Save jump to avoid weird recalculations
+            jump: (0.125*Math.pow(2, i-1)),
+            group: this.currentDeck,
+            inKey: "beatjump_"+(0.125*Math.pow(2, i-1))+"_forward",
+            outKey: "beatjump_"+(0.125*Math.pow(2, i-1))+"_forward",
```

Comment:
there is no reason to specify both explicitly. If `inKey === outKey`, just specifing `key` is enough (except when in shift/unshift functions).